app/models/chat_channel.rb

```diff
@@ -91,6 +91,10 @@ def find_or_create_chat_channel(channel_type, slug, contrived_name)
       end
       channel
     end
+
+    def invalid_direct_channel?(users, channel_type)
+      (users.size != 2 || users.map(&:id).uniq.count < 2) && channel_type == "direct"
```

Comment:
Should we split up the errors? Maybe create an error (in `./app/errors/` or in the model) like `InvalidDirectChannel`.

And then change the logic to:
```ruby
raise InvalidDirectChannel, "Users are the same"
raise InvalidDirectChannel, "Can't have more than 2 users in a direct channel"
```

Reply:
Since this happens so infrequently I think leaving it is fine for now.